src/main.test.ts

```diff
@@ -0,0 +1,433 @@
+    mock.inputs['current-status'] = 'failure';
+
+    await run();
+
+    expect(mockCore.setFailed).toHaveBeenCalledTimes(0);
+    expect(mockCore.info).toHaveBeenCalledWith('Current run status: failure');
+    expect(mockCore.info).toHaveBeenCalledWith('No notification needed');
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Last run status: completed/failure',
+    );
+  });
+
+  it('not a pull_request: should send success notification if last run failed and current succeed', async () => {
+    github.context.payload.pull_request = undefined;
+    await run();
+    expect(mockCore.setFailed).toHaveBeenCalledTimes(0);
+    expect(mockCore.info).toHaveBeenCalledWith('Current run status: success');
+    expect(mockCore.info).toHaveBeenCalledWith('Success notification');
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Last run status: completed/failure',
+    );
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Message body: {"username":"workflow-status-slack-notification CI alert","icon_emoji":":bangbang:","attachments":[{"color":"good","author_name":"workflowactor","author_link":"https://github.com/workflowactor","author_icon":"https://github.com/workflowactor.png?size=32","fields":[{"title":"Repository","value":"workflow-status-slack-notification","short":true},{"title":"Branch","value":"main","short":true},{"title":"Action URL","value":"<https://github.com/reside-eng/workflow-status-slack-notification/actions/runs/23456|Publish Action>","short":true},{"title":"Event","value":"pull_request","short":true},{"title":"Publish Action workflow success","value":"Previously failing Publish Action workflow in workflow-status-slack-notification succeed.","short":false}]}]}',
+    );
+  });
+});
+
+// Test using GH CLI to retrieve previous status (new commit workflow behavior)
+describe('last run status retrieved from GH CLI (new commit workflow behavior)', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    mock = {
+      // Default action inputs
+      inputs: {
+        'current-status': 'success',
+        'slack-webhook': `${slackUrl}${slackPath}`,
+        'github-token': `${process.env.GITHUB_TOKEN}`,
+      },
+      repo: {
+        owner: 'reside-eng',
+        repo: 'workflow-status-slack-notification',
+      },
+      workflow: 'Failure workflow (for test purpose only)',
+      runId: 23456,
+      headRef: 'main',
+      ref: 'main',
+      eventName: 'pull_request',
+      actor: 'workflowactor',
+      serverUrl: 'https://github.com',
+    };
+
+    // All tests will work under the AUT directory
+    process.env.GITHUB_WORKSPACE = join(process.cwd(), 'aut');
+
+    mockCore.getInput.mockImplementation(
+      (name: string): string => mock.inputs[name] || '',
+    );
+
+    mockCache.restoreCache.mockImplementation(
+      (
+        paths: string[],
+        primaryKey: string,
+        restoreKeys?: string[] | undefined,
```

Comment:
Are we using these or can they be removed?

Reply:
These parameters are not used in the mocked method but I thought that for a mock the method signature should be the same as the method that is mocked. I tried to remove these parameters and I have test errors, I think that it is not able to retrieve the mock when the method is called because the parameters are different.
Unless there is something I'm missing here ?

Reply:
Ok. That makes sense. I was seeing some unused var but if it causes an issue, don't worry about it.

Reply:
If we pull it into a lib of its own following the TODO, then it would make sense to test at that time.